storage-node/packages/cli/src/commands/upload.ts

```diff
@@ -0,0 +1,207 @@
+
+            return dataObject.unwrap();
+        } catch (err) {
+            this.fail(`Cannot create data object: ${err}`);
+        }
+    }
+
+    // Uploads file to given asset URL.
+    private async uploadFile(assetUrl: string) {
+        // Create file read stream and set error handler.
+        const file = fs.createReadStream(this.mediaSourceFilePath)
+            .on('error', (err) => {
+                this.fail(`File read failed: ${err}`);
+            });
+
+        // Upload file from the stream.
+        try {
+            const fileSize = this.getFileSize();
+            const config: AxiosRequestConfig = {
+                headers: {
+                    'Content-Type': '', // https://github.com/Joystream/storage-node-joystream/issues/16
+                    'Content-Length': fileSize.toString()
+                },
+                'maxContentLength': MAX_CONTENT_LENGTH,
+            };
+            await axios.put(assetUrl, file, config);
+
+            console.log("File uploaded.");
+        } catch (err) {
+            this.fail(err.toString());
+        }
+    }
+
+    // Requests the runtime and obtains the storage node endpoint URL.
+    private async discoverStorageProviderEndpoint(storageProviderId: string): Promise<string> {
+        try {
+            const serviceInfo = await discover(storageProviderId, this.api);
+
+            if (serviceInfo === null) {
+                this.fail("Storage node discovery failed.");
+            }
+            debug(`Discovered service info object: ${serviceInfo}`);
+
+            const dataWrapper = JSON.parse(serviceInfo);
+            const assetWrapper = JSON.parse(dataWrapper.serialized);
+
+            return assetWrapper.asset.endpoint;
+        } catch (err) {
+            this.fail(`Could not get asset endpoint: ${err}`);
+        }
+    }
+
+    // Loads and unlocks the runtime identity using the key file and pass phrase.
+    private async loadIdentity(): Promise<any> {
+        try {
+            await fs.promises.access(this.keyFile);
+        } catch (error) {
+            this.fail(`Cannot read file "${this.keyFile}".`)
+        }
+
+        return this.api.identities.loadUnlock(this.keyFile, this.passPhrase);
+    }
+
+    // Shows command usage. Overrides the abstract method from the base class.
+    protected showUsage() {
+        console.log(chalk.yellow(`
+        Usage:   storage-cli upload mediaSourceFilePath dataObjectTypeId keyFilePath passPhrase memberId
+        Example: storage-cli upload ./movie.mp4 1 ./keyFile.json secretPhrase 1
+      `));
+    }
+
+    // Command executor.
+    async run() {
+        // Checks for input parameters, shows usage if they are invalid.
+        if (!this.assertParameters()) return;
+
+        let addContentParams = await this.getAddContentParams();
+        debug(`AddContent Tx params: ${JSON.stringify(addContentParams)}`);
+        debug(`Decoded CID: ${ContentId.decode(addContentParams.contentId).toString()}`);
```

Comment:
see my earlier comment about how content id is stored in runtime.

```suggestion
        debug(`Decoded CID: ${addContentParams.contentId.encode()}`);
```

Reply:
Ok